src/main/java/uk/gov/hmcts/reform/hmc/data/HearingEntity.java

```diff
@@ -183,4 +188,24 @@ public Integer getNextRequestVersion() {
     public boolean hasHearingResponses() {
         return getHearingResponses() != null && !getHearingResponses().isEmpty();
     }
+
+    public HearingEntity updateLastGoodStatus() {
+        HearingStatus currentStatus = this.getStatus() != null
+            ? HearingStatus.valueOf(this.getStatus()) : null;
+        HearingStatus lastGoodStatus = this.getLastGoodStatus() != null
+            ? HearingStatus.valueOf(this.getLastGoodStatus()) : null;
+
+        if (lastGoodStatus != null && lastGoodStatus != currentStatus) {
+            if (HearingStatus.isFinalStatus(lastGoodStatus)) {
+                throw new BadRequestException("Status is in a Final State" + currentStatus);
```

Comment:
should I throw an exception or silently fail and allow final status to be updated since its not needed for functionality